sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -231,6 +234,21 @@ CosmosClientBuilder setApiType(ApiType apiType){
         return this;
     }
 
+    /**
+     * Adds a policy for modifying request options dynamically.
+     *
+     * @param policy the policy to add
+     * @return current cosmosClientBuilder
+     */
+    public CosmosClientBuilder addPolicy(CosmosOperationPolicy policy) {
```

Comment:
maybe addRequestOptionsPolicy - or addOperationPolicy or addOpwrationSettingsPolicy?- just policy seems way too vague?

Reply:
in general I am torn - my preference would be to add not more public surface area using the term RequestOptions - those are all not request (transport) options but operation scoped settings - on the other hand we use RequestOptions already in other places so for consistency sake keep using it seems also a reasonable approach

Reply:
Changed to CosmosOperationPolicy. I think that it is fine changing here to using operation, but for the CosmosRequestOptions, it is better to leave it as request options to be consistent with the rest .